src/mono/System.Private.CoreLib/System.Private.CoreLib.csproj

```diff
@@ -361,6 +361,9 @@
     <IsSourceProject>true</IsSourceProject>
     <ApiCompatValidateBaseline>true</ApiCompatValidateBaseline>
   </PropertyGroup>
+  <ItemGroup>
+    <MatchingRefApiCompatBaseline Condition="'$(TargetsWindows)' != 'true'" Include="MatchingRefApiCompatBaseline.Unix.txt" />
```

Comment:
In the past, we never allowed API differences based on the platform that the assembly is targeting. Just wanted to bring this up.

Why not baseline the error for all platforms?

cc @ericstj

Reply:
We have it in the shared baseline today: https://github.com/dotnet/runtime/blob/main/src/libraries/System.IO.FileSystem/src/MatchingRefApiCompatBaseline.txt#L8 . Would be consistent at least.

Reply:
Yeah the problem is that this doesn't work in conjunction with the ApiCompatValidateBaseline setting (which ensures the baseline isn't containing unnecessary suppressions) since the `System.IO.Enumeration.FileSystemEntry.<_fileNameBuffer>e__FixedBuffer` only shows up on Unix platforms as it is the only one using the fixed array field: https://github.com/dotnet/runtime/blob/e415018a152a4a2e0db7619ceb91a3b3326b217a/src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEntry.Unix.cs#L20

I talked to Roslyn and it's by design that the backing type is public even though the field is private (since the type is unspeakable so you can't actually do anything with it from C#) so I was planning on updating ApiCompat to ignore it.

The other option would be to turn off ApiCompatValidateBaseline.

Reply:
https://github.com/dotnet/runtime/pull/73182 should make this problem go away.

Reply:
Cool yeah that's a better option.